server/app/services/export/JsonExporterService.java

```diff
@@ -115,15 +118,20 @@ public DocumentContext convertApplicationExportDataToJsonArray(
   }
 
   private ApplicationExportData buildApplicationExportData(
-      ApplicationModel application, ProgramDefinition programDefinition) {
-    ReadOnlyApplicantProgramService roApplicantProgramService =
-        applicantService.getReadOnlyApplicantProgramService(application, programDefinition);
+      ApplicationModel application,
+      ImmutableList<ProgramDefinition> programDefinitionsForAllVersions) {
 
-    ImmutableList<AnswerData> answerDataList =
-        roApplicantProgramService.getSummaryDataAllQuestions();
-    ImmutableMap.Builder<Path, Optional<?>> entriesBuilder = ImmutableMap.builder();
+    Map<Path, AnswerData> uniqueAnswers = new HashMap<>();
 
-    for (AnswerData answerData : answerDataList) {
+    for (ProgramDefinition pd : programDefinitionsForAllVersions) {
+      applicantService
+          .getReadOnlyApplicantProgramService(application, pd)
+          .getSummaryDataAllQuestions()
+          .forEach(data -> uniqueAnswers.putIfAbsent(data.contextualizedPath(), data));
```

Comment:
I thought about leaving a comment explaining this so lmk if that'd be helpful:
These are all working with the same `Application`, and therefore the same `ApplicantData` object. We're just using the questions on each program to find a path into that `ApplicantData` object to look for an answer.

Since the path is immutable for a given question, we don't really care which version of a question we get, just that we get all the possible questions. That said, we do happen to be encountering the newest questions first.

Reply:
Actually I just added the comment

Reply:
Thanks!